File: Challenges/LowCostBiggerRightCount/biggerThanLeft.java
package Challenges.LowCostBiggerRightCount;

public class biggerThanLeft {
    public static void main(String[] args) {
        int[] a = {1, 14, 5, 2, 5};

        func(a);
    }

    static void func(int[] a) {
        int[] b = new int[a.length];

        for (int i = 0; i < a.length; i++) {
            int contAtual = 0;

            for (int j = i; j < a.length; j++) {

                if (a[j] > a[i]) {
                    contAtual++;
                }
            }
            b[i] = contAtual;
        }

        for (int i : b) {
            System.out.println(i);
        }
    }

}

File: DSA_C/Challenges/BiggerThanLeft.c
#include <stdio.h>
#include "BiggerThanLeft.h"


int main() {
    int a[] = {1, 14, 5, 2, 5};

    func(a, sizeof(a)/sizeof(a[0]));
    return 0;
}

void func(int a[], int n) {
    int b[n];

    for (int i = 0; i < n; i++) {
        int contAtual = 0;

        for (int j = i; j < n; j++) {

            if (a[j] > a[i]) {
                contAtual++;
            }
        }
        b[i] = contAtual;
    }

    for (int i = 0 ; i < n; i++) {
        printf("%d ", b[i]);
    }
}
